scripts/only-allow-pnpm.js

```diff
@@ -13,8 +13,8 @@ function pmFromUserAgent(userAgent) {
   const pmSpec = userAgent.split(' ')[0]
   const separatorPos = pmSpec.lastIndexOf('/')
   return {
-    name: pmSpec.substr(0, separatorPos),
-    version: pmSpec.substr(separatorPos + 1),
+    name: pmSpec.substring(0, separatorPos),
```

Comment:
I guess I'd probably prefer using `slice` here as well for consistency (or, alternatively, `substring` everywhere – they have the same arguments anyway in all of these cases).

Reply:
`slice` and `substring` are very similar but [differ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#differences_between_substring_and_slice) in how they handle the second parameter, especially if it's negative. I used it here to have the same result as `substr`